code/script.js

```diff
@@ -218,108 +234,193 @@ const generateBoard = () => {
+    value: questions.value,
+  };
+};
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+// Function that displays how many questions player has asked
+const showTimesAsked = (number) => {
+  document.getElementById(
+    "questionCounter"
+  ).innerHTML = `Questions asked: ${number}`;
+};
 
-  currentQuestion = {
-    category: category,
-    // value: value
+// This is the function that will count time and will be executed with setInterval() method
+const timeCounter = () => {
+  timerDisplay.innerHTML = `${min} min : ${sec} sec`;
+  sec++;
+  if (sec === 60) {
+    sec = 0;
+    min++;
   }
-}
+};
 
 // This function should be invoked when you click on 'Find Out' button.
+// Function displays a different text depending on the category and calls a function to filter characters
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  const { category, value } = currentQuestion;
+  numberOfQuestionsAsked++;
+  showTimesAsked(numberOfQuestionsAsked);
+  if (category === "hair") {
+    if (value === secret.hair) {
+      alert(
+        `Yes, this person has ${value} hair. We keep all people with ${value} hair`
+      );
+      filterCharacters(true);
+    } else {
+      alert(
+        `No, this person doesn't have ${value} hair. We remove all people with ${value} hair`
+      );
+      filterCharacters();
+    }
+  } else if (category === "eyes") {
+    if (value === secret.eyes) {
+      alert(
+        `Yes, this person has ${value} eyes. We keep all people with ${value} eyes`
+      );
+      filterCharacters(true);
+    } else {
+      alert(
+        `No, this person doesn't have ${value} eyes. We remove all people with ${value} eyes`
+      );
+      filterCharacters();
+    }
+  } else if (category === "accessories") {
+    if (secret.accessories.includes(value)) {
+      alert(
+        `Yes, this persons wears ${value}. We keep all people with ${value}`
+      );
+      filterCharacters(true);
+    } else {
+      alert(
+        `No, this person doesn't wear ${value}. We remove all people with ${value}`
+      );
+      filterCharacters();
+    }
+  } else {
+    if (secret.other.includes(value)) {
+      alert(
+        `Yes, this persons is a ${value}. We keep all people who are ${value}`
+      );
+      filterCharacters(true);
+    } else {
+      alert(
+        `No, this persons isn't a ${value}. We remove all people who are ${value}`
+      );
+      filterCharacters();
+    }
   }
```

Comment:
I like the fact that you merged two functions here: showing the alert in the same function when you compare the question values with the stored secret value. Nice touch